R/nist.R

```diff
@@ -170,76 +331,27 @@ tidy_ritable <- function(ri_xml) {
-                             "thickness" = "Phase thickness (\u03bcm)",
-                             "temp_start" = "Tstart (C)",
-                             "temp_end" = "Tend (C)",
-                             "temp_rate" = "Heat rate (K/min)",
-                             "hold_start" = "Initial hold (min)",
-                             "hold_end" = "Final hold (min)",
-                             "reference" = "Reference",
-                             "comment" = "Comment") %>%
-        # fix column types and make uniform contents of some columns
-        dplyr::mutate_at(
-          dplyr::vars(
-            "length",
-            "diameter",
-            "thickness",
-            "temp_start",
-            "temp_end",
-            "temp_rate",
-            "hold_start",
-            "hold_end",
-            "RI"
-          ),
-          as.numeric
-        )
-
-    } else if (temp_prog == "isothermal") {
-      tidy2 <- dplyr::select(tidy1,
-                             "RI" = "I",
-                             "type" = "Column type",
-                             "phase" = "Active phase",
-                             "length" = "Column length (m)",
-                             "gas" = "Carrier gas",
-                             "substrate" = "Substrate",
-                             "diameter" = "Column diameter (mm)",
-                             "thickness" = "Phase thickness (\u03bcm)",
-                             "temp" = "Temperature (C)",
-                             "reference" = "Reference",
-                             "comment" = "Comment") %>%
-        # fix column types and make uniform contents of some columns
-        dplyr::mutate_at(vars("length", "diameter", "thickness", "temp",  "RI"),
-                  as.numeric)
-    }
+    tidy2 <- dplyr::select(tidy1,
+                           "RI" = "I",
+                           "column" = "Column type",
+                           "phase" = "Active phase",
+                           "length" = "Column length (m)",
+                           "gas" = "Carrier gas",
+                           "substrate" = "Substrate",
+                           "diameter" = "Column diameter (mm)",
+                           "thickness" = "Phase thickness (\u03bcm)",
+                           "program" = contains("Program"),
+                           "temp" = contains("Temperature (C)"),
+                           "temp_start" = contains("Tstart (C)"),
+                           "temp_end" = contains("Tend (C)"),
+                           "temp_rate" = contains("Heat rate (K/min)"),
+                           "hold_start" = contains("Initial hold (min)"),
+                           "hold_end" = contains("Final hold (min)"),
+                           "reference" = "Reference",
+                           "comment" = "Comment") %>%
+  dplyr::mutate_at(vars(any_of(c("length", "diameter", "thickness", "temp",
```

Comment:
I _think_ you don't need `vars()` anymore with dplyr 
```suggestion
  dplyr::mutate_at(any_of(c("length", "diameter", "thickness", "temp",
```

Reply:
It looks like the `vars()` actually is still required. I am going to revert this commit

Reply:
`mutate_at()` is superseded, but it's fine to just leave it for now.  The "new" pattern is something like 
```
mutate(across(any_of(c("length", "diameter", "thickness", ...)),  as.numeric))
```